packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js

```diff
@@ -0,0 +1,320 @@
+`;
+
+const listStyle = css`
+  margin-block-start: ${({ theme }) => theme.spaces[4]};
+  margin-block-end: ${({ theme }) => theme.spaces[4]};
+  margin-inline-start: ${({ theme }) => theme.spaces[0]};
+  margin-inline-end: ${({ theme }) => theme.spaces[0]};
+  padding-inline-start: ${({ theme }) => theme.spaces[4]};
+
+  ol,
+  ul {
+    margin-block-start: ${({ theme }) => theme.spaces[0]};
+    margin-block-end: ${({ theme }) => theme.spaces[0]};
+  }
+`;
+
+const Orderedlist = styled.ol`
+  list-style-type: decimal;
+  ${listStyle}
+`;
+
+const Unorderedlist = styled.ul`
+  list-style-type: disc;
+  ${listStyle}
+`;
+
+const List = ({ attributes, children, element }) => {
+  if (element.format === 'ordered') return <Orderedlist {...attributes}>{children}</Orderedlist>;
+
+  return <Unorderedlist {...attributes}>{children}</Unorderedlist>;
+};
+
+List.propTypes = {
+  attributes: PropTypes.object.isRequired,
+  children: PropTypes.node.isRequired,
+  element: PropTypes.shape({
+    format: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+const Img = styled.img`
+  max-width: 100%;
+`;
+
+const Image = ({ attributes, children, element }) => {
+  const { url, alternativeText, width, height } = element.image;
+
+  return (
+    <Box {...attributes}>
+      {children}
+      <Box contentEditable={false}>
+        <Img src={url} alt={alternativeText} width={width} height={height} />
+      </Box>
+    </Box>
+  );
+};
+
+Image.propTypes = {
+  attributes: PropTypes.object.isRequired,
+  children: PropTypes.node.isRequired,
+  element: PropTypes.shape({
+    image: PropTypes.shape({
+      url: PropTypes.string.isRequired,
+      alternativeText: PropTypes.string,
+      width: PropTypes.number,
+      height: PropTypes.number,
+    }).isRequired,
+  }).isRequired,
+};
+
+/**
+ * Manages a store of all the available blocks.
+ *
+ * @returns {{
+ *   [key: string]: {
+ *     renderElement: (props: Object) => JSX.Element,
+ *     variants: {
+ *       [key: string]: {
+ *         icon: React.ComponentType,
+ *         label: {id: string, defaultMessage: string},
+ *         value: Object,
+ *         matchNode: (node: Object) => boolean,
+ *       }
+ *     }
+ *   }
+ * }} an object containing rendering functions and metadata for different blocks, indexed by name.
+ */
+export function useBlocksStore() {
+  return {
+    paragraph: {
+      renderElement: (props) => (
+        <Typography as="p" variant="omega" {...props.attributes}>
+          {props.children}
+        </Typography>
+      ),
+      variants: {
+        paragraph: {
+          icon: Code,
+          label: {
+            id: 'components.Blocks.blocks.text',
+            defaultMessage: 'Text',
+          },
+          value: {
+            type: 'paragraph',
+          },
+          matchNode: (node) => node.type === 'paragraph',
+        },
+      },
+    },
+    heading: {
+      renderElement: (props) => <Heading {...props} />,
+      variants: {
+        'heading-one': {
```

Comment:
Instead of heading with variants couldn't we just flatten? Each heading be it's own object, with it's own renderElement heading? Now we have the idea of heading and heading has a level and in addition we have heading and heading has variants.

Reply:
The reason for this structure was that the variants share many things from their parent. But I agree about all the downsides you are pointing out. I'll see what it looks like without variants 👍

Reply:
I gave it a try and I do think it's better, so I just pushed the changes (last commit). Let me know what you think